programmers/난이도별/level01.모의고사/sjjyy.java

```diff
@@ -0,0 +1,121 @@
+package 난이도별.level01.모의고사;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class sjjyy {
+    public static int[] solution(int[] answers) {
+        int[] ans1 = {1, 2, 3, 4, 5};
+        int[] ans2 = {2, 1, 2, 3, 2, 4, 2, 5};
+        int[] ans3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5,};
+        int a = 0, b = 0, c = 0;
+
+        int len = answers.length;
+
+        for (int i = 0; i < len; i++) {
+            if (answers[i] == ans1[i % 5])
+                a++;
+            if (answers[i] == ans2[i % 8])
+                b++;
+            if (answers[i] == ans3[i % 10])
+                c++;
+        }
+
+        int max = Math.max(a, Math.max(b, c));
+
+        List<Integer> list = new ArrayList<>();
+        if (max == a)
+            list.add(1);
+        if (max == b)
+            list.add(2);
+        if (max == c)
+            list.add(3);
+
+        int[] answer = new int[list.size()];
+
+        for (int i = 0; i < list.size(); i++) {
+            answer[i] = list.get(i);
+        }
+
+        return answer;
+    }
+
+    public static int[] solution2(int[] answers) {
+        int[] ans1 = {1, 2, 3, 4, 5};
+        int[] ans2 = {2, 1, 2, 3, 2, 4, 2, 5};
+        int[] ans3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5,};
+        int [] score = new int[3];
+
+        int len = answers.length;
+
+        for (int i = 0; i < len; i++) {
+            if (answers[i] == ans1[i % 5])
+                score[0]++;
+            if (answers[i] == ans2[i % 8])
+                score[1]++;
+            if (answers[i] == ans3[i % 10])
+                score[2]++;
+        }
+
+        int max = Math.max(score[0], Math.max(score[1], score[2]));
+
+        List<Integer> list = new ArrayList<>();
+        if (max == score[0])
+            list.add(1);
+        if (max == score[1])
+            list.add(2);
+        if (max == score[2])
+            list.add(3);
+
+        int[] answer = new int[list.size()];
+
+        for (int i = 0; i < list.size(); i++) {
+            answer[i] = list.get(i);
+        }
+
+        return answer;
+    }
+
+    public static int[] solution3(int[] answers) {
+        int[] ans1 = {1, 2, 3, 4, 5};
+        int[] ans2 = {2, 1, 2, 3, 2, 4, 2, 5};
+        int[] ans3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5,};
```

Comment:
```suggestion
        List<Integer> ans = Arrays.asList(
            Arrays.asList(1, 2, 3, 4, 5),
            Arrays.asList(2, 1, 2, 3, 2, 4, 2, 5),
            Arrays.asList(3, 3, 1, 1, 2, 2, 4, 4, 5, 5),
        );
```
2차원 배열 혹은 리스트로 만들면 어떨까요?

Reply:
asList로는 새로운 배열 객체를 만들 수 없는 것 같아서 (이부분은 제가 잘 몰라서 틀렸다면 알려주세요!) 2차원 배열로 수정했습니다 😃

Reply:
2차원 List
```java
    public static void main(String[] args) {
        List<List<Integer>> answers = Arrays.asList(
            Arrays.asList(1, 2, 3, 4),
            Arrays.asList(11, 22, 33, 44),
            Arrays.asList(111, 222, 333, 444)
        );

        System.out.println(answers);
    }
```

혹은 list안에 int[]

```java
public class JunilHwang {
    public static void main(String[] args) {
        List<int[]> answers = Arrays.asList(
            new int[]{1, 2, 3, 4},
            new int[]{11, 22, 33, 44},
            new int[]{111, 222, 333, 44}
        );

        System.out.println(answers);
    }
}
```

이렇게 만들 수 있어요!